lms/validation/_helpers/_base.py

```diff
@@ -45,6 +42,40 @@ def parse(self, *args, **kwargs):
         return _PYRAMID_PARSER.parse(self, self.context["request"], *args, **kwargs)
 
 
+class RequestsResponseSchema(_BaseSchema):
+    """Base class for schemas that validate ``requests`` lib responses."""
+
+    def __init__(self, response):
+        super().__init__()
+        self.context = {"response": response}
+
+    def parse(self, *args, **kwargs):
+        """
+        Parse and validate the response.
+
+        Use this schema to parse and validate ``self.context["response"]`` and
+        either return the successfully parsed params or raise a validation
+        error.
+
+        :raise lms.validation.ValidationError: if the response isn't valid
+        """
+        response = self.context["response"]
+
+        try:
+            response_params = response.json()
+        except ValueError as err:
+            raise ValidationError(
+                {"_schema": "response doesn't have a valid JSON body"}
```

Comment:
The `"_schema"` key is marshmallow's standard way of handling an error that doesn't apply to any one particular field

Reply:
My reading of the `ValidationError` API docs is that `_schema` is the default field, so `ValidationError(<string message>)` would be an equivalent way of writing this?

Reply:
Ah, except this is an `lms.validation.ValidationError` not a `marshmallow.ValidationError`. I've moved the JSON parsing into a marshmallow pre-load method so that it can make use of this `marshmallow.ValidationError` feature